ocs_ci/helpers/performance_lib.py

```diff
@@ -375,3 +380,210 @@ def csi_bulk_pvc_time_measure(interface, pvc_objs, operation, start_time):
+                            results[name]["delete"][
+                                "start"
+                            ] = extruct_timestamp_from_log(line)
+                        if re.search(f'delete "{pv_name}": succeeded', line):
+                            results[name]["delete"]["end"] = extruct_timestamp_from_log(
+                                line
+                            )
+                            results[name]["delete"]["time"] = calculate_operation_time(
+                                name, results[name]["delete"]
+                            )
+
+    # Getting times from CSI log - if needed
+    del_pv_names = []
+    for i in range(0, len(pvc_name)):
+        del_pv_names.append("")
+
+    if csi_logs:
+        for sublog in csi_logs:
+            for line in sublog:
+                for i in range(0, len(pvc_name)):
+                    name = pvc_name[i].name
+                    pv_name = pvc_name[i].backed_pv
+
+                    if "generated volume id" in line.lower() and pv_name in line:
+                        del_pv_names[i] = line.split("(")[1].split(")")[0]
+                    if op in ["all", "create"]:
+                        if f"Req-ID: {pv_name} GRPC call:" in line:
+                            results[name]["csi_create"][
+                                "start"
+                            ] = extruct_timestamp_from_log(line)
+                        if f"Req-ID: {pv_name} GRPC response:" in line:
+                            results[name]["csi_create"][
+                                "end"
+                            ] = extruct_timestamp_from_log(line)
+                            results[name]["csi_create"][
+                                "time"
+                            ] = calculate_operation_time(
+                                name, results[name]["csi_create"]
+                            )
+                    if op in ["all", "delete"]:
+                        if del_pv_names[i]:
+                            if f"Req-ID: {del_pv_names[i]} GRPC call:" in line:
+                                results[name]["csi_delete"][
+                                    "start"
+                                ] = extruct_timestamp_from_log(line)
+                            if f"Req-ID: {del_pv_names[i]} GRPC response:" in line:
+                                results[name]["csi_delete"][
+                                    "end"
+                                ] = extruct_timestamp_from_log(line)
+                                results[name]["csi_delete"][
+                                    "time"
+                                ] = calculate_operation_time(
+                                    name, results[name]["csi_delete"]
+                                )
+
+    logger.debug(f"All results are : {json.dumps(results, indent=3)}")
+    return results
+
+
+def wait_for_resource_bulk_status(
+    resource, resource_count, namespace, status, timeout=60, sleep_time=3
```

Comment:
Can you use `TimeoutSampler`?

Reply:
i try to avoid using object to reduce memory consumption as posible.